tests/seed_isort_config_test.py

```diff
@@ -95,12 +99,72 @@ def test_integration_extra_file(tmpdir):
         assert tmpdir.join('.isort.cfg').read() == expected
 
 
-def test_integration_no_config(tmpdir, capsys):
+@pytest.mark.parametrize(
+    'initial_filesystem,expected_filesystem',
+    [
+        ({}, {'.isort.cfg': '[settings]\nknown_third_party=cfgv\n'}),
+        (
+            {'setup.cfg': '[bdist_wheel]\nuniversal = True\n'},
+            {
+                'setup.cfg': '[bdist_wheel]\nuniversal = True\n\n'
+                '[isort]\nknown_third_party=cfgv\n',
+            },
+        ),
+        (
+            {'tox.ini': '[tox]\nenvlist = py36\n'},
+            {
+                'tox.ini': (
+                    '[tox]\nenvlist = py36\n\n'
+                    '[isort]\nknown_third_party=cfgv\n'
+                ),
+            },
+        ),
+        (
+            {
+                'setup.cfg': '[bdist_wheel]\nuniversal = True\n',
+                'tox.ini': '[tox]\nenvlist = py36\n',
+            },
+            {
+                'setup.cfg': '[bdist_wheel]\nuniversal = True\n',
+                'tox.ini': (
+                    '[tox]\nenvlist = py36\n\n'
+                    '[isort]\nknown_third_party=cfgv\n'
+                ),
+            },
+        ),
+    ],
+)
+def test_integration_no_section(
+    tmpdir,
+    initial_filesystem,
+    expected_filesystem,
+):
     with tmpdir.as_cwd():
         tmpdir.join('f.py').write('import cfgv')
+        for filename, initial in initial_filesystem.items():
+            tmpdir.join(filename).write(initial)
         _make_git()
 
-        assert main(())
+        assert not main(())
+
+        for filename, expected in expected_filesystem.items():
+            assert tmpdir.join(filename).read() == expected
+
+
+def test_integration_has_section_without_config(tmpdir):
+    with tmpdir.as_cwd():
+        tmpdir.join('.isort.cfg').write(
+            '[settings]\n'
+            'combine_as_imports=true\n',
+        )
+        tmpdir.join('f.py').write('import cfgv')
+        _make_git()
 
-        out, _ = capsys.readouterr()
-        assert out.startswith('Could not find a `known_third_party` setting')
+        assert not main(())
+
+        expected = (
+            '[settings]\n'
+            'known_third_party=cfgv\n'
+            'combine_as_imports=true\n'
```

Comment:
this also presents one of the reasons I didn't want to just add the setting, config files are usually a _conscious_ decision about ordering, etc. -- as written it potentially de-alphabetizes or reorders where the setting is

Reply:
Makes sense.. let's continue in the other thread..